tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -1113,13 +1118,35 @@ def _run_pseudolabeling(self, unlabeled_data: pd.DataFrame, max_iter: int,
                 X_test = X_test.loc[test_pseudo_idxes[~test_pseudo_idxes].index]
                 previous_score = current_score
 
+        # Should be done at the end. When done at each iteration caused the model to get much worse
+        if fit_ensemble:
+            logger.log(15, 'Fitting weighted ensemble using models trained with pseudo labeled data')
+
+            if return_pred_prob:
+                logger.log(30, "WARNING: 'return_pred_prob' is set to True and 'fit_ensemble' is set to true."
+                               " Fit ensemble is done at the end of training so predictions returned are not"
+                               " done by the model ensembling models trained using pseudo labeled data.")
+
+            weighted_ensemble_model_name = self.fit_weighted_ensemble()[0]
+
+            # TODO: This is a hack! self.predict_prob does not update to use weighted ensemble
+            # if it's the best model.
+            # TODO: There should also be PL added to weighted ensemble model name to notify
+            # users it is a model trained with PL models if they are indeed ensembled
+            # TODO: Maybe transductive setting should have the WE model predict and return that?
+            # I found that had very good results on covertype. Needs to be tested on AMBL
+            model_best_name = self._trainer.leaderboard().iloc[0]['model']
+            if model_best_name == weighted_ensemble_model_name:
+                self._trainer.model_best = model_best_name
+                self._trainer.save()
+
```

Comment:
should we print an informative statement when the new ensemble is not the best? In that case, the user may assume the new ensemble is the one being used for prediction even when it is not.

Reply:
added log